processing/src/main/java/io/druid/segment/data/VSizeIndexedIntsWriter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.google.common.io.ByteStreams;
+import com.google.common.io.CountingOutputStream;
+import com.google.common.primitives.Ints;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * Streams integers out in the binary format described by VSizeIndexedInts
+ */
+public class VSizeIndexedIntsWriter extends SingleValueIndexedIntsWriter
+{
+  private static final byte version = VSizeIndexedInts.version;
+
+  private final IOPeon ioPeon;
+  private final String filenameBase;
+  private final int numBytes;
+
+  private CountingOutputStream valuesOut = null;
+
+  public VSizeIndexedIntsWriter(
+      final IOPeon ioPeon,
+      final String filenameBase,
+      final int maxValue
+  )
+  {
+    this.ioPeon = ioPeon;
+    this.filenameBase = filenameBase;
+    this.numBytes = VSizeIndexedInts.getNumBytesForMax(maxValue);
+  }
+
+  @Override
+  public void open() throws IOException
+  {
+    valuesOut = new CountingOutputStream(ioPeon.makeOutputStream(makeFilename("values")));
+  }
+
+  @Override
+  protected void addValue(int val) throws IOException
+  {
+    byte[] intAsBytes = Ints.toByteArray(val);
+    valuesOut.write(intAsBytes, intAsBytes.length - numBytes, numBytes);
```

Comment:
since we use only last "numBytes" from intAsBytes, does it make sense to not create full 4 byte intAsBytes array?